### PR TITLE
Btm 357 move event names to config repo

### PR DESCRIPTION
**Pull request by Si3rr4wow**

### CODE REVIEW

**Comment by Si3rr4wow** [comment]
Looks like this needs more mocking. One for Friday morning me thinks

File: src/handlers/int-test-support/helpers/envHelper.ts [hunk 1]
```diff
@@ -8,3 +8,12 @@ export const configStackName = (): string => `di-btm-cfg-${configName()}`;
 
 export const runViaLambda = (): boolean =>
   process.env.TEST_VIA_LAMBDA === "true";
+
+export const outputQueue = (): string => {
+  if (
+    process.env.OUTPUT_QUEUE_URL === undefined ||
```
Comment (phinterseer) [inline src/handlers/int-test-support/helpers/envHelper.ts:14]: I'd do this check at the beginning of the handler like in other lambdas.

Also this is a test-helper, so would be good to have another helper in actual production code for this.

File: src/handlers/int-test-support/helpers/envHelper.ts [hunk 2]
```diff
@@ -8,3 +8,12 @@ export const configStackName = (): string => `di-btm-cfg-${configName()}`;
...
+    process.env.OUTPUT_QUEUE_URL.length === 0
```
Comment (phinterseer) [inline src/handlers/int-test-support/helpers/envHelper.ts:15]: And while we're at it, we should refactor our handlers to have an init-phase which is triggered on first function start-up as described [here](https://serverlessfirst.com/function-initialisation/). We could also load the required configs in this init-functions. @gdsgj basically did almost the same thing in his caching approach already and it seems to be a accepted pattern in Lambda to use global code to do some init-stuff. I'd prefer they'd explicitly call an init()-function on startup, but I can live with this approach. :-)

File: src/shared/utils/config-utils/fetch-event-names.test.ts [hunk 3]
```diff
@@ -0,0 +1,21 @@
+import { fetchEventNames } from "./fetch-event-names";
+
+jest.mock("./fetch-vendor-service-config", () => ({
+  fetchVendorServiceConfig: () => [
+    { event_name: "BOUGHT_EGGS" },
+    { event_name: "MADE_CAKE" },
+    { event_name: "ATE_CAKE" },
+    { event_name: "MADE_CAKE" },
+    { event_name: "MADE_TEA" },
+    { event_name: "ATE_CAKE" },
```
Comment (phinterseer) [inline src/shared/utils/config-utils/fetch-event-names.test.ts:10]: Is there a failure case with 'HAVE_CAKE' and 'EAT_CAKE' somewhere? ;-)

File: src/shared/utils/config-utils/fetch-vendor-service-config.test.ts [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+    mockedCsvConverterFromString = jest.fn(() => mockedVendorServiceConfig);
+    mockedCsvConverter = { fromString: mockedCsvConverterFromString };
+    mockedGetCsvConverter.mockReturnValue(mockedCsvConverter);
+  });
+
+  test("Throws error if no vendor service config found", async () => {});
+
+  test("Throws error if vendor service config is not valid", async () => {});
+
+  test("Converts CSV to Json", async () => {});
```
Comment (phinterseer) [inline src/shared/utils/config-utils/fetch-vendor-service-config.test.ts:63]: This looks slightly unfinished.

File: src/handlers/int-test-support/helpers/envHelper.ts (hunk 2 above)
Comment (Si3rr4wow) [reply src/handlers/int-test-support/helpers/envHelper.ts:15]: Yeah I was thinking similar things, seems like every handler starts with a block of if this throw that, would be nice to tidy that up.

File: src/handlers/int-test-support/helpers/envHelper.ts (hunk 1 above)
Comment (Si3rr4wow) [reply src/handlers/int-test-support/helpers/envHelper.ts:14]: Oops, I should have checked the directory name 🫢

File: src/shared/utils/config-utils/fetch-vendor-service-config.test.ts (hunk 4 above)
Comment (Si3rr4wow) [reply src/shared/utils/config-utils/fetch-vendor-service-config.test.ts]: Filled these in, thanks for catching

File: src/shared/utils/config-utils/fetch-event-names.test.ts (hunk 3 above)
Comment (Si3rr4wow) [reply src/shared/utils/config-utils/fetch-event-names.test.ts:10]: Hmm I tried it but somehow the test still passes 🤔

File: src/handlers/int-test-support/helpers/envHelper.ts (hunk 2 above)
Comment (Si3rr4wow) [reply src/handlers/int-test-support/helpers/envHelper.ts]: I've taken this helper out and moved the check over to the handler. I'll have a think about an init function refactor

**Review by MelAtGDS — APPROVED** [review]

**Pull request merged**